lock-api/src/main/java/com/palantir/lock/client/CoalescingLeaderTimeGetter.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import com.palantir.common.concurrent.CoalescingSupplier;
+import com.palantir.lock.v2.LeaderTime;
+import java.util.function.Supplier;
+
+public final class CoalescingLeaderTimeGetter implements LeaderTimeGetter {
```

Comment:
non-actionable: This is fine. I'm not super happy with the name just because the other intended implementation is also functionally coalescing, but don't have a better solution.